include/boost/geometry/strategies/geographic/area_geographic.hpp

```diff
@@ -0,0 +1,212 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2016 Oracle and/or its affiliates.
+// Contributed and/or modified by Vissarion Fisikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_AREA_GEOGRAPHIC_HPP
+#define BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_AREA_GEOGRAPHIC_HPP
+
+#include <boost/geometry/formulas/area_formulas.hpp>
+#include <boost/geometry/formulas/thomas_inverse.hpp>
+//#include <boost/math/special_functions/atanh.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace area
+{
+
+/*!
+\brief Geographic area calculation by trapezoidal rule plus integral
+approximation that gives the ellipsoidal correction
+
+}
+
+*/
+
+template
+<
+    typename PointOfSegment,
+    typename Strategy = void,
+    const std::size_t SeriesOrder = 2,
+    bool ExpandEpsN = true,
+    bool LongSegment = false,
+    typename Spheroid = void,
+    typename CalculationType = void
+>
```

Comment:
`Spheroid` is not needed here. `spheroid_constants` have `CT` type anyway. The constructor could take `Spheroid` as template parameter.

`const` keyword is not needed here.

Now the most tricky part.

- `Strategy` is not a strategy in the Boost.Geometry meaning so it should be renamed if we decided to pass it. It'd rather be something like `Policy` (we have several) or `InverseFormula`.
- Inverse solution formulas take `bool` parameters enabling various parts of computation so by taking the formula type we're relying on the user to create proper type, i.e. enabling azimuths computation. So because of that, and for convenience instead of a type `typename InverseFormula` a template could be required here `template <typename, bool, bool, bool, bool ,bool> class InverseFormula`.
- I'm not sure if it's a good idea to allow/force the user to pass formulas into strategies. I'd rather hide them (i.e. not expose in the "official" interface) and use only internally, i.e. treat them like defined in `detail` namespace. This way we could safely change their interface if needed.

So I see two possibilities:
1. Implement several strategies: area_andoyer, area_thomas, area_vincenty
2. Take only `size_t ApproximationOrder` or some `typename ApproximationTag`. Then in compile time instantiate corresponding inverse formula and use corresponding order of series expansion. As a side note, the problem with `ApproximationOrder` is that we could have several methods using the same order. And `ApproximationTag` would be less convenient for the user than solution 1.